packages/package-store/src/storeController/createImportPackage.ts

```diff
@@ -16,58 +15,61 @@ import renameOverwrite = require('rename-overwrite')
 import { promisify } from 'util'
 import linkIndexedDir from '../fs/linkIndexedDir'
 
-const execFilePromise = promisify(child_process.execFile)
 const ncp = promisify(ncpCB)
 const limitLinking = pLimit(16)
 
-export default (packageImportMethod?: 'auto' | 'hardlink' | 'copy' | 'reflink'): ImportPackageFunction => {
+export default (packageImportMethod?: 'auto' | 'hardlink' | 'copy' | 'clone'): ImportPackageFunction => {
   const importPackage = createImportPackage(packageImportMethod)
   return (filesResponse, dependency, opts) => limitLinking(() => importPackage(filesResponse, dependency, opts))
 }
 
-function createImportPackage (packageImportMethod?: 'auto' | 'hardlink' | 'copy' | 'reflink') {
-  let fallbackToCopying = false
+function createImportPackage (packageImportMethod?: 'auto' | 'hardlink' | 'copy' | 'clone') {
+  let auto = async (
```

Comment:
This can be its own function in the enclosing module scope.

I’d put it immediately after the `createImportPackage(…)` function.

Reply:
This must be inside the scope of createImportPackage because I am reassigning the auto variable inside the function. If I move it out, it won't have access to the auto variable.

Reply:
Do you need to assign to the `auto` variable?

Reply:
There might be a better way but this is what I want. When the first file is imported from the store, this function will try to clone or hardlink it. It detects what the filesystem allows and next time it will not try to clone, if cloning is not supported, the hardlinking or copying function will be called for the rest of installation